### PR TITLE
feat(console): integrate language settings

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

We can switch language in Admin Console now.

Apr-30-2022 09-39-02

## Linear Issue Reference

[LOG-2266](https://linear.app/silverhand/issue/LOG-2266/adminconsoles-language-config)

## Testing

- [x] Switch language in Settings page, the application switches to selected language

### CODE REVIEW

File: packages/console/src/i18n/init.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
     .use(initReactI18next)
     .use(LanguageDetector)
-    .init({
-      resources,
-      fallbackLng: 'en',
-      interpolation: {
-        escapeValue: false,
-      },
-    });
+    .init(language ? { ...baseOptions, lng: language } : baseOptions);
```
[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/console/src/i18n/init.ts:18]: actually in this case no `baseOptions` is needed

**Review by simeng-li — APPROVED** [review]
overall LGTM please address the comments before merge.

File: packages/console/src/i18n/init.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (demonzoo) [reply packages/console/src/i18n/init.ts]: Yeah... Late night code sometimes DOES look weird. 😂

File: packages/console/src/App.tsx [hunk 2]
```diff
@@ -53,6 +53,12 @@ const Main = () => {
     };
   }, [data?.adminConsole.appearanceMode, defaultTheme]);
 
+  useEffect(() => {
+    (async () => {
+      void initI18n(data?.adminConsole.language);
+    })();
+  }, [data?.adminConsole.language]);
```
[the remark this answers is not in the document]
Comment (demonzoo) [reply packages/console/src/App.tsx:60]: Done. [LOG-2341](https://linear.app/silverhand/issue/LOG-2341/display-loading-when-switching-languages-in-ac)

**Review by gao-sun — APPROVED** [review]

**Pull request merged**